### DISCUSSION TITLE
could i use one script as library/include_file in another script?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#17963] [Q&A] [answered]
I made a script with functions for simple automation scenes. Than i made a script for the webgui for webbuttons to trigger this automation scenes. but this not working as i expected.

If the webbutton is pushed i want to execute a external functions.

web.be create and handle the webbuttons and tests with print("button pushed") work well.

functions.be has the functions test() that works fine from function.be

load("functions.be") inside web.be do run the script functions.be one time but i cant use the functions from function.be inside web.be

is there a way  to do this?
Or do i have to use driver or have to write both scripts in one?
Or can i share a global var to the loaded script "functions.be"?
thx!

**Comment by sfromis** [comment]
One strategy could be to create your library as a Berry module, which you then can import in other Berry code. Using import does not execute any code, but makes the library available in a similar way as when you do a `import string` and so on.

If you *do* need the library to be initialized, and/or hold some cross-call internal state, you could still have it as a module, where you call an initialization function returning a class instance. That class could be internal to the module, not available outside of the intended access strategy via the class instance. All the methods of that class could be used by any code having the instance reference in a variable, in a similar way as when using a class like `webclient` or `bytes`.

**Comment by ChrisVonFrosta** [comment]
could you show me an entry point to create a module? it seems that i am very bad in googling berry things.

**Comment by ChrisVonFrosta** [comment, marked as answer]
maybe i found it?!
https://berry.readthedocs.io/en/latest/source/en/Chapter-8.html
![grafik](https://user-images.githubusercontent.com/16272493/219528769-1d0f625f-9215-4d78-9ed9-1343e3a58a66.png)
